locker/src/main/java/org/killbill/commons/locker/zookeeper/ZooKeeperGlobalLockDao.java

```diff
@@ -0,0 +1,107 @@
+import org.killbill.commons.locker.zookeeper.recipes.LockListener;
+import org.killbill.commons.locker.zookeeper.recipes.WriteLock;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ZooKeeperGlobalLockDao implements GlobalLockDao {
+
+    private static final Logger LOG = LoggerFactory.getLogger(ZooKeeperGlobalLockDao.class);
+    private static final List<ACL> DEFAULT_ACL = ZooDefs.Ids.OPEN_ACL_UNSAFE;
+
+    // TODO jgomez: we can read this property from the external properties
+    private static final String zkPath = "/kb/locks/accounts/";
+
+    private final String name;
+    private final String path;
+    private final WriteLock writeLock;
+    private final CountDownLatch lockAcquiredSignal = new CountDownLatch(1);
+
+    public ZooKeeperGlobalLockDao(final String lockName, final Long timeout) {
+        this.name = lockName;
+        this.path = zkPath + lockName;
+        ZooKeeper zooKeeper = null;
+        try {
+            // TODO jgomez: what should we use as the connectString value? Using localhost by now
+            zooKeeper = new ZooKeeper("localhost", timeout.intValue(), new Watcher() {
+                @Override
+                public void process(final WatchedEvent event) {
+                    if (event.getState() == Event.KeeperState.SyncConnected) {
+                        lockAcquiredSignal.countDown();
+                    }
+                }
+            });
+            lockAcquiredSignal.await();
+        } catch (Exception e) {
+            // TODO jgomez catch and throw corresponding exceptions
+        }
+        this.writeLock = new WriteLock(zooKeeper, path, DEFAULT_ACL, new SyncLockListener());
+    }
+
+    class SyncLockListener implements LockListener {
+        @Override
+        public void lockAcquired() {
+            LOG.debug("Lock acquired by {} on {}", name, path);
+            lockAcquiredSignal.countDown();
+        }
+
+        @Override
+        public void lockReleased() {
+            LOG.debug("Lock released by {} on {}", name, path);
+        }
+    }
+
+    @Override
+    public boolean lock(final Connection connection, final String lockName, final long timeout, final TimeUnit timeUnit) throws SQLException {
+        try {
+            LOG.debug("{} requesting lock on {}...", lockName, path);
+            writeLock.lock();
+            return lockAcquiredSignal.await(timeout, timeUnit);
+        } catch (Exception e) {
+            // TODO jgomez catch and throw corresponding exceptions
+            return false;
+        }
+    }
+
+    @Override
+    public boolean releaseLock(final Connection connection, final String lockName) throws SQLException {
+        // TODO jgomez: implement! Is this the unlock method?
+        writeLock.unlock();
+        return false;
+    }
+
+    @Override
+    public boolean isLockFree(final Connection connection, final String lockName) throws SQLException {
+        // TODO jgomez: implement! How to check this in the writeLock?
```

Comment:
Either the `WriteLock`class already offers such method -- did not verify -- or we will need to implement one. Most of the logic should already be in place though in this class

Reply:
WriteLock has a ´isOwner()´ method that says if the current Node is the owner of the Lock.
Is that what we are expecting here?

Reply:
@javier-gomez-github  : Don't implement it for now. We can come back to it later